packages/cache/src/internal/cacheUtils.ts

```diff
@@ -0,0 +1,99 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as glob from '@actions/glob'
+import * as io from '@actions/io'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as util from 'util'
+import {v4 as uuidV4} from 'uuid'
+import {CacheFilename, CompressionMethod} from './constants'
+
+// From https://github.com/actions/toolkit/blob/master/packages/tool-cache/src/tool-cache.ts#L23
+export async function createTempDirectory(): Promise<string> {
+  const IS_WINDOWS = process.platform === 'win32'
+
+  let tempDirectory: string = process.env['RUNNER_TEMP'] || ''
```

Comment:
If the env variable isn't set, is the temp directory expected to be created in the current working directory? There is an [extra check](https://github.com/actions/toolkit/blob/36e732155e521874637327a845b689ddc7a20645/packages/tool-cache/src/tool-cache.ts#L557) in the `tool-cache` package

Reply:
See below, if `tempDIrectory` is empty we create our own temp directory. Note this is an edge case, as `RUNNER_TEMP` should always be defined.